src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs

```diff
@@ -563,6 +597,27 @@ async Task<PurgeHistoryResult> IDurableOrchestrationClient.PurgeInstanceHistoryA
             return await this.DurabilityProvider.PurgeInstanceHistoryByInstanceId(instanceId);
         }
 
+        async Task<PurgeHistoryResult> IDurableOrchestrationClient.PurgeInstanceHistoryAsync(IEnumerable<string> instanceId)
+        {
+            var tasks = new List<Task>();
+            int instanceDeleted = 0;
+            foreach (string id in instanceId)
+            {
+                tasks.Add(Task.Run(async () =>
+                {
+                    var purgeHistoryResult = await this.DurabilityProvider.PurgeInstanceHistoryByInstanceId(id);
+                    if (purgeHistoryResult.InstancesDeleted == 1)
+                    {
+                        instanceDeleted += 1;
+                    }
+                }));
+            }
+
+            await Task.WhenAll(tasks);
```

Comment:
There's a much simpler way to do this. You can use a parallel for each loop instead of manually using tasks.

Also, make sure the int for instancesDeleted is incremented in a way that is safe for multiple threads to write to it.